applications/minotari_merge_mining_proxy/src/block_template_protocol.rs

```diff
@@ -122,59 +126,23 @@ impl BlockTemplateProtocol<'_> {
                 );
                 (data, height)
             } else {
-                let (new_template, block_template_with_coinbase, height) = match block_templates
-                    .get_new_template(best_block_hash)
-                    .await
-                {
+                let block = match self.p2pool_client.as_mut() {
+                    Some(client) => {
+
+                        let block_result = client
+                            .get_new_block(GetNewBlockRequest::default())
+                            .await?
+                            .into_inner();
+                        let new_block_result = block_result.block.ok_or_else(|| MmProxyError::FailedToGetBlockTemplate("block result".to_string()))?;
```

Comment:
Have you used somewhere the returning target difficulty?
Just because now p2pool does not modifies the original difficulty in miner_data.